@xen-orchestra/proxy/src/app/mixins/backups/index.js

```diff
@@ -233,5 +235,68 @@ export default class Backups {
         ],
       },
     })
+
+    const getPartition = disposable(function* ({ disk, partition, remote }) {
+      const adapter = yield app.remotes.getAdapter(remote)
+      return yield adapter.getPartition(disk, partition)
+    })
+
+    const partitionDisposers = {}
+    const dispose = async () => {
+      await Promise.all(
+        Object.keys(partitionDisposers).map(path => {
+          const disposers = partitionDisposers[path]
+          delete partitionDisposers[path]
+          return Promise.all(disposers.map(d => d(path).catch(noop)))
+        })
+      )
+      app.hooks.removeListener('stop', dispose)
+    }
+    app.hooks.on('stop', dispose)
```

Comment:
https://devdocs.io/node~10_lts/events#events_emitter_once_eventname_listener

```suggestion
    }
    app.hooks.once('stop', dispose)
```